### ISSUE TITLE
Debugger stuck on waiting for connection from debug service on chrome when trying to run on web using Flutter 2.x

### ISSUE THREAD

**Issue by felipefranca824** [Dart-Code/Dart-Code#4370]
**Describe the bug**
Debugger stuck on waiting for connection from debug service on chrome when trying to run on web

**To Reproduce**
1. having a flutter project with an older version 2.10.5 for example
2. click on debug
![image](https://user-images.githubusercontent.com/42588746/215763950-146afdc8-c480-413c-9007-bf58e3c2b8bf.png)

**Expected behavior**
project was supposed to run normally

**Screenshots**
![image](https://user-images.githubusercontent.com/42588746/215764435-78b9d673-444e-47dc-866c-01ef5805b639.png)

**Please complete the following information:**
 - Operating System and version: Ubuntu 22.04.1 LTS
 - VS Code version: 1.74.3
 - Dart extension version: 3.58.0
 - Dart/Flutter SDK version: 2.16.2/2.10.5
 - Target device (if the issue relates to Flutter debugging): chrome

**Comment by DanTup** [comment]
@felipefranca824 from the error in your screenshot, it looks like there is a problem with the code (trying to use `??` after a non-nullable variable).

Does fixing that issue resolved the problem?

If not, can you produce a small sample project that triggers this issue that you can share? Thanks!

**Comment by felipefranca824** [comment]
Reproduced in a new project
![image](https://user-images.githubusercontent.com/42588746/216358651-8dcbebb8-00f9-4edc-ab8f-861f6de559a7.png)
doesn't come out of it.

but if i run by command line "fllutter run -d chome" it works

**Comment by DanTup** [comment]
@felipefranca824 can you reproduce this reliably?

Did you make any changes to the sample project, or run it as is (and how did you start it?).

Could you try capturing debugging logs while reproducing it (using the sample project)?

- Run the [**Dart: Capture Debugging Logs** command](https://dartcode.org/docs/logging/#capture-debugging-logs)
- Run the project to reproduce the issue
- Wait at least 30s after Chrome appears, just to ensure the log isn't cut short of anything that might be running slowly
- Click **Cancel** on the logging notification to stop logging and open the log
- Remove anything sensitive from the log
- Attach the log here, in a [gist](https://gist.github.com/), or email to me at logs@dartcode.org.

Thanks!

**Comment by felipefranca824** [comment]
@DanTup 
I ran the project without any modifications.

I sent the log file to your email

**Comment by DanTup** [comment]
@felipefranca824 thanks! I found the issue in your log:

```
setIsolatePauseMode: NoSuchMethodError: Class 'ChromeProxyService' has no instance method 'setIsolatePauseMode' with matching arguments.
Receiver: Instance of 'ChromeProxyService'
Tried calling: setIsolatePauseMode(\"1\", exceptionPauseMode: \"Unhandled\", shouldPauseOnExit: null)
Found: setIsolatePauseMode(String, {String exceptionPauseMode, bool shouldPauseOnExit}) => Future<Success>
```

This was a bug in earlier versions of Flutter (https://github.com/dart-lang/webdev/issues/1627). It's since been fixed but you'll need to upgrade Flutter to a more recent version to get the fix (the current version is 3.7).

The reason it works from the command line is that this doesn't set up the debugger, which avoids the code that is failing.

Please let me know if you still have the issue after upgrading!

**Comment by felipefranca824** [comment]
After updating the problem was solved.

to continue using in older versions of flutter I used version 3.56.0 of the dartcode extension.

thank you

**Comment by psimondk** [comment]
I'm getting this issue revisiting a 3 flutter web project there hasn't been touched for 7 months - so very likely one of the latest dart extension upgrades is the culprit - its very annoying and i do not really wanna do a 2.x 3.x upgrade as we're close to go-live for a large client

**Comment by DanTup** [comment]
@psimondk the quickest fix would be rolling back the Dart extension to v3.56:

<img width="504" alt="Screenshot 2023-02-13 at 11 29 12" src="https://user-images.githubusercontent.com/1078012/218446263-081c47a0-d061-4c4f-9034-b3fc85be1805.png">

I'll have a look to see whether there's a version we can gate on that avoids this issue for both web/non-web.

**Comment by DanTup** [comment]
I've updated the version for when the extension start using the new API to something I believe works for both cases. If you switch to the Pre-release channel for the extension, you'll get this fix now without needing to use an older version of the extension:

<img width="756" alt="Screenshot 2023-02-13 at 12 19 54" src="https://user-images.githubusercontent.com/1078012/218456057-b695fd12-3db0-471e-acd8-4c267993680b.png">

You can switch back to the stable versions of the extension after the next release (or, stick with this - it should generally be stable anyway, as it's usually only published to make fixes like this available to people a little sooner).

**Comment by psimondk** [comment]
And I can confirm that it works like a charm - thanks a lot for stepping up
to help on this - we had a very good demo for a German Client last friday
and now we can debug our accompanying dashboard too!

[image: image.png]

Best Regards, Palle S.

**Comment by lbevilacqua** [comment]
Hello,

I encountered the issue too on a Flutter 2.10.5 project.
Using the Dart extension pre-release version solved it.

Thanks

**Comment by sperochon** [comment]
Hello,

If it can help someone...

I had the same issue on a Flutter 2.10 project using Android Studio and Flutter plugin (version 73).
As suggested by previous comments, I have installed previous release of Flutter plugin (version 72) manually and it is now working fine.

Here are the steps :

1. I have downloaded the zip plugin file here (be careful to select "compatibility" to "Android Studio"):
https://plugins.jetbrains.com/plugin/9212-flutter/versions/stable

2. Then installed the file manually as described here :
https://stackoverflow.com/a/17358862/1836827

And it works :-)
